presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplitManager.java

```diff
@@ -155,7 +156,16 @@ public RaptorSplitSource(
             this.effectivePredicate = requireNonNull(effectivePredicate, "effectivePredicate is null");
             this.transactionId = requireNonNull(transactionId, "transactionId is null");
             this.bucketToNode = requireNonNull(bucketToNode, "bucketToNode is null");
-            this.iterator = new SynchronizedResultIterator<>(shardManager.getShardNodes(tableId, bucketed, merged, effectivePredicate));
+
+            ResultIterator<BucketShards> iterator;
+            if (bucketed) {
+                checkArgument(bucketToNode.isPresent(), "bucketToNode not present for bucketed table");
+                iterator = shardManager.getShardNodesBucketed(tableId, merged, bucketToNode.get(), effectivePredicate);
```

Comment:
It actually calls `getNodeShards` which gets the shards for a node. This is getting the nodes for shards (or shard/node pairs).